pySDC/projects/PinTSimE/battery_implicit.py

```diff
@@ -0,0 +1,160 @@
+
+
+def main(dt=1e-2, use_switch_estimator=True):
+    """
+    A simple test program to do SDC/PFASST runs for the battery drain model
+    """
+
+    # initialize level parameters
+    level_params = dict()
+    level_params['restol'] = 1e-8
+    level_params['dt'] = dt
+
+    # initialize sweeper parameters
+    sweeper_params = dict()
+    sweeper_params['quad_type'] = 'LOBATTO'
+    sweeper_params['num_nodes'] = 5
+    sweeper_params['initial_guess'] = 'zero'
+
+    # initialize problem parameters
+    problem_params = dict()
+    problem_params['newton_maxiter'] = 200
+    problem_params['newton_tol'] = 1e-08
+    problem_params['Vs'] = 5.0
+    problem_params['Rs'] = 0.5
+    problem_params['C'] = 1.0
+    problem_params['R'] = 1.0
+    problem_params['L'] = 1.0
+    problem_params['alpha'] = 1.2
+    problem_params['V_ref'] = 1.0
+    problem_params['set_switch'] = np.array([False], dtype=bool)
+    problem_params['t_switch'] = np.zeros(1)
+
+    # initialize step parameters
+    step_params = dict()
+    step_params['maxiter'] = 10
+
+    # initialize controller parameters
+    controller_params = dict()
+    controller_params['logger_level'] = 20
+    controller_params['hook_class'] = log_data
+
+    # convergence controllers
+    switch_estimator_params = {}
+    convergence_controllers = {SwitchEstimator: switch_estimator_params}
+
+    # fill description dictionary for easy step instantiation
+    description = dict()
+    description['problem_class'] = battery_implicit  # pass problem class
+    description['problem_params'] = problem_params  # pass problem parameters
+    description['sweeper_class'] = generic_implicit  # pass sweeper
+    description['sweeper_params'] = sweeper_params  # pass sweeper parameters
+    description['level_params'] = level_params  # pass level parameters
+    description['step_params'] = step_params
+
+    if use_switch_estimator:
+        description['convergence_controllers'] = convergence_controllers
+
+    proof_assertions_description(description, problem_params)
+
+    # set time parameters
+    t0 = 0.0
+    Tend = 0.5
+
+    # instantiate controller
+    controller = controller_nonMPI(num_procs=1, controller_params=controller_params, description=description)
+
+    # get initial values on finest level
+    P = controller.MS[0].levels[0].prob
+    uinit = P.u_exact(t0)
+
+    # call main function to get things done...
+    uend, stats = controller.run(u0=uinit, t0=t0, Tend=Tend)
+
+    Path("data").mkdir(parents=True, exist_ok=True)
+    fname = 'data/battery_implicit.dat'
+    f = open(fname, 'wb')
+    dill.dump(stats, f)
+    f.close()
+
+    # filter statistics by number of iterations
+    iter_counts = get_sorted(stats, type='niter', recomputed=False, sortby='time')
+
+    # compute and print statistics
+    min_iter = 20
+    max_iter = 0
+
+    f = open(
+        'data/battery_implicit_out_dt{}_USE{}_{}.txt'.format(dt, use_switch_estimator, description['sweeper_class']),
```

Comment:
Yeah, that's not a good filename.  It breaks the CI because if contains `<` and `>`. Did you mean `.__name__`, maybe?